Sage-Bionetworks/schematic
# Change `Uuid` column name to `Id`

## What Changed
This PR makes the switch from naming the (former) `Uuid` column containing UUID values to `Id` instead.

### Manifests
Manifests that contain columns named `Uuid` will have the column name updated to `Id` at upload, and the `.csv` file will be saved with the new convention, **overwriting the version with the old convention**.

### Tables
Tables that have `Uuid` columns will have the column name updated at the next manifest upload whether using the `replace` or `upsert` table operations. All new tables created going forward will be using the `Id` naming convention. Functionality is included to handle cases where manifests and tables have mismatched naming conventions; any adhering to the old convention will be updated to the new `Id` convention.

Tested with example data model manifests and HTAN model manifests

## To Test

1. `git checkout develop`
2. Generate a **new** manifest. If a manifest already exists in the dataset folder then delete and regenerate
    * `Biospecimen` for HTAN entity based, table creation or replacement
        * Can use dataset `syn51398208`
    *  `ScRNASeqLevel1` for HTAN file based, table creation or replacement
        * Can use dataset `syn51398808`
    *  `MockRDB` for example data model, table upserts
        * Can use dataset `syn51470395`
4. If using provided datasets in project `syn27042964`, then delete `Biospecimen`, `ScRNA-SeqLevel1`, and `MockRDB` tables
5. Submit as table with `--table_manipulation` set appropriately
6. `git checkout develop-ID-column-FDS-70` 
7. Add row to manifest; do not change column names
8. Submit as table with `--table_manipulation` set appropriately
9. Verify table reflects all rows in the manifest, with the `Uuid` column now named `Id`
10. Verify local copy of the manifest has had the `Uuid` column changed to be named `Id`
